### PR TITLE
Read and write KHR_materials_common extension

### PR DESCRIPTION
**Pull request by Mavhrin**
Not sure about the name used to create the structure which holds the properties defined by KHR_materials_common: Common

### CODE REVIEW

File: code/glTF2Asset.h [hunk 1]
```diff
@@ -174,7 +175,8 @@ namespace glTF2
 	#define AI_MATKEY_GLTF_ALPHAMODE "$mat.gltf.alphaMode", 0, 0
 	#define AI_MATKEY_GLTF_ALPHACUTOFF "$mat.gltf.alphaCutoff", 0, 0
 	#define AI_MATKEY_GLTF_PBRSPECULARGLOSSINESS "$mat.gltf.pbrSpecularGlossiness", 0, 0
-	#define AI_MATKEY_GLTF_PBRSPECULARGLOSSINESS_GLOSSINESS_FACTOR "$mat.gltf.pbrMetallicRoughness.glossinessFactor", 0, 0
+	#define AI_MATKEY_GLTF_PBRSPECULARGLOSSINESS_GLOSSINESS_FACTOR "$mat.gltf.pbrSpecularGlossiness.glossinessFactor", 0, 0
+    #define AI_MATKEY_GLTF_COMMON "$mat.gltf.common", 0, 0
```
Comment (jblorteau) [inline code/glTF2Asset.h:179]: Extra space at line start (many more below), is that normal?

File: code/glTF2Asset.h [hunk 2]
```diff
@@ -742,6 +748,26 @@ namespace glTF2
 		void SetDefaults();
     };
 
+    struct Common
```
Comment (jblorteau) [inline code/glTF2Asset.h:751]: `CommonMaterial` might be more expressive, but they use `pbrSpecularGlossiness` as well so... might be better to keep it consistent with how it is already in the codebase.

File: code/glTF2AssetWriter.inl [hunk 3]
```diff
@@ -572,17 +621,30 @@ namespace glTF2 {
...
         {
-            // This is used to export pbrSpecularGlossiness materials with GLTF 2.
-            if (this->mAsset.extensionsUsed.KHR_materials_pbrSpecularGlossiness) {
-                exts.PushBack(StringRef("KHR_materials_pbrSpecularGlossiness"), mAl);
+            for (Asset::IdMap::iterator it = this->mAsset.extensionsUsed.begin(), itEnd = this->mAsset.extensionsUsed.end(); it!=itEnd; ++it) {
+                if (it->second) {
+                  extsUsed.PushBack(StringRef(it->first), mAl);
+                }
+            }
+        }
```
Comment (jblorteau) [inline code/glTF2AssetWriter.inl:632]: Is there a use for these braces? I don't see how a block is used here. (Just curious, apparently it's already there)

File: code/glTF2Exporter.cpp [hunk 4]
```diff
@@ -413,48 +414,62 @@ void glTF2Exporter::ExportMaterials()
...
-                normalizedShininess = normalizedShininess * specularIntensity;
-
-                m->pbrMetallicRoughness.roughnessFactor = 1 - normalizedShininess;
+        
+        // The value is true if the material is from common material source or the export option is true
+        bool hasCommonSource = false;
+        mat->Get(AI_MATKEY_GLTF_COMMON, hasCommonSource);
+        bool exportAsCommonMaterial = hasCommonSource || (mProperties->HasPropertyBool(AI_CONFIG_EXPORT_GLTF2_MATERIAL_COMMON) && mProperties->GetPropertyBool(AI_CONFIG_EXPORT_GLTF2_MATERIAL_COMMON));
+        // The value is false if the export as a common material is active and the export option is false
+        bool exportAsPBRMaterial = !(exportAsCommonMaterial && mProperties->HasPropertyBool(AI_CONFIG_EXPORT_GLTF2_MATERIAL_PBR) && !mProperties->GetPropertyBool(AI_CONFIG_EXPORT_GLTF2_MATERIAL_PBR));
```
Comment (jblorteau) [inline code/glTF2Exporter.cpp:423]: Isn't there an issue if `HasPropertyBool(AI_CONFIG_EXPORT_GLTF2_MATERIAL_PBR)` is false? Shouldn't it be true for the `GetProperty` to be safe or is there a default?

**Review by jblorteau — APPROVED** [review]

File: code/glTF2Asset.h (hunk 1 above)
Comment (Mavhrin) [reply code/glTF2Asset.h:179]: There are mixed whitespaces in the code using 4 spaces or just a tab. In XCode and Visual Code they are rendered the same way but not in github.

File: code/glTF2Asset.h (hunk 2 above)
Comment (Mavhrin) [reply code/glTF2Asset.h:751]: Yes Common is just a poor naming. It's weird when it is used in other functions. I don't know what to do.

File: code/glTF2AssetWriter.inl (hunk 3 above)
Comment (Mavhrin) [reply code/glTF2AssetWriter.inl:632]: Yep, I kept the existing syntax. The external braces have no utility.

File: code/glTF2Exporter.cpp (hunk 4 above)
Comment (Mavhrin) [reply code/glTF2Exporter.cpp:423]: The default value for AI_CONFIG_EXPORT_GLTF2_MATERIAL_PBR  is True (I added the description in config.h.in). If I can't guarantee that only the KHR_materials_common is needed by explicitly deactivating the pbr export and activating the extension, the pbr is exported.

**Pull request merged**